crates/bevy_reflect/src/type_uuid_impl.rs

```diff
@@ -0,0 +1,88 @@
+impl_type_uuid!(usize, "0129e1d8cff041f9b23aa99c6e1006b8");
+impl_type_uuid!(i8, "af7a5411661e43b0b1631ea43a825fd2");
+impl_type_uuid!(i16, "68592d5de5be4a608603c6988edfdf9c");
+impl_type_uuid!(i32, "439ff07f96c94aa5a86352ded71e4730");
+impl_type_uuid!(i64, "7f9534793ad24ab2b9f05d8254f4204a");
+impl_type_uuid!(i128, "6e5009be5845460daf814e052cc9fcf0");
+impl_type_uuid!(isize, "d3d52630da45497faf86859051c79e7d");
+impl_type_uuid!(f32, "006607124a8148e1910c86f0c18c9015");
+impl_type_uuid!(f64, "a5bc32f5632b478c92a0939b821fff80");
+impl_type_uuid!(<T,E> Result, "d5960af2e8a743dfb7427dd59b70df95");
+impl_type_uuid!(String, "c9f90d31b52d4bcd8b5c1d8b6fc1bcba");
+impl_type_uuid!(PathBuf, "aa79933abd1743698583a3acad3b8989");
+impl_type_uuid!(<T> Vec, "ab98f5408b974475b643662247fb3886");
+impl_type_uuid!(<K, V> HashMap,"f37bfad9ca8c4f6ea7448f1c39e05f98");
+impl_type_uuid!(<T> Option, "8d5ba9a9031347078955fba01ff439f0");
+#[cfg(feature = "smallvec")]
+impl_type_uuid!(
+    <T: smallvec::Array> smallvec::SmallVec,
+    "26fd5c1bed7144fbb8d1546c02ba255a"
+);
+impl_type_uuid!(<K> HashSet, "5ebd2379ece44ef2b1478262962617a3");
+impl_type_uuid!(<T> RangeInclusive, "79613b729ca9490881c7f47b24b22b60");
+impl_type_uuid!(<T> RangeFrom, "1bd8c975f122486c9ed443e277964642");
+impl_type_uuid!(<T> RangeTo, "7d938903749a4d198f496cb354929b9b");
+impl_type_uuid!(<T> RangeToInclusive, "2fec56936206462fa5f35c99a62c5ed1");
+impl_type_uuid!(RangeFull, "227af17f65db448782a2f6980ceae25d");
+impl_type_uuid!(Duration, "cee5978c60f74a53b6848cb9c46a6e1c");
+impl_type_uuid!(Instant, "9b0194a1d31c44c1afd2f6fd80ab8dfb");
+impl_type_uuid!(NonZeroI128, "915a1e7fcaeb433982cebf58c2ac20e7");
+impl_type_uuid!(NonZeroU128, "286de521146042cda31dfbef8f3f6cdc");
+impl_type_uuid!(NonZeroIsize, "9318740a9fd14603b709b8fbc6fd2812");
+impl_type_uuid!(NonZeroUsize, "a26533ed16324189878263d5e7a294ce");
+impl_type_uuid!(NonZeroI64, "1aa38623127a42419cca4992e6fc3152");
+impl_type_uuid!(NonZeroU64, "46be65e669a2477d942e2ec39d0d2af7");
+impl_type_uuid!(NonZeroU32, "cf53a46d9efe4022967160cb61762c91");
+impl_type_uuid!(NonZeroI32, "a69fbd659bef4322b88b15ff3263f530");
+impl_type_uuid!(NonZeroI16, "8744c2ec8a10491fae40f8bafa58b30d");
+impl_type_uuid!(NonZeroU16, "c7b8b60780a6495bab4fda2bdfedabcc");
+impl_type_uuid!(NonZeroU8, "635ee104ef7947fb9d7f79dad47255a3");
+impl_type_uuid!(NonZeroI8, "2d3f1570b7f64779826d44da5c7ba069");
+#[cfg(any(unix, windows))]
+impl_type_uuid!(OsString, "809e7b3c1ea240979ecd832f91eb842a");
+
+macro_rules! impl_tuple {
+    ( $($name: ident)+ ) => {
+        impl_type_uuid!(< $($name),+ > ( $($name),+ ), "35c8a7d3d4b34bd7b8471118dc78092f");
+    };
+}
+
+impl_tuple!(A B);
+impl_tuple!(A B C);
+impl_tuple!(A B C D);
+impl_tuple!(A B C D E);
+impl_tuple!(A B C D E F);
+impl_tuple!(A B C D E F G);
+impl_tuple!(A B C D E F G H);
+impl_tuple!(A B C D E F G H I);
+impl_tuple!(A B C D E F G H I J);
+impl_tuple!(A B C D E F G H I J K);
+impl_tuple!(A B C D E F G H I J K L);
```

Comment:
why not use `bevy::ecs::all_tuples`?

```suggestion
macro_rules! impl_tuple {
    ( $($name: ident),* ) => {
        // privately scope `Tuple` type.
        const _: () = {
            type Tuple< $($name),* > = ( $($name,)* ); // e.g. `Tuple<A, B> = (A, B,)`
            impl_type_uuid!(< $($name),* > Tuple, "35c8a7d3d4b34bd7b8471118dc78092f");
        };
        
    };
}

bevy::ecs::all_tuples!(impl_tuple, 0, 12);
```

this would also work for the unit tuple `()` and the single-element tuple `(A,)` which you seem to have missed out.